tests/checkpointing/test_client_module.py

```diff
@@ -96,3 +96,74 @@ def test_client_checkpointer_module(tmp_path: Path) -> None:
+        pre_aggregation=pre_aggregation_checkpointer, post_aggregation=post_aggregation_checkpointer
+    )
+
+    model_1 = LinearTransform()
+    model_2 = LinearTransform()
+
+    checkpoint_module.maybe_checkpoint(model_1, 0.78, {"test_1": 1.0}, CheckpointMode.PRE_AGGREGATION)
+    checkpoint_module.maybe_checkpoint(model_2, 0.78, {"test_1": 1.0}, CheckpointMode.POST_AGGREGATION)
+
+    assert checkpoint_module.pre_aggregation is not None
+    assert checkpoint_module.post_aggregation is not None
+
+    loaded_pre_model_best = pre_aggregation_checkpointer[0].load_best_checkpoint()
+    loaded_pre_model_latest = pre_aggregation_checkpointer[1].load_best_checkpoint()
+    loaded_post_model = post_aggregation_checkpointer.load_best_checkpoint()
+
+    assert isinstance(loaded_pre_model_best, LinearTransform)
+    # pre aggregation model should be the same as model_1
+    assert torch.equal(model_1.linear.weight, loaded_pre_model_best.linear.weight)
+
+    assert isinstance(loaded_pre_model_latest, LinearTransform)
+    # pre aggregation model should be the same as model_1
+    assert torch.equal(model_1.linear.weight, loaded_pre_model_latest.linear.weight)
+
+    assert isinstance(loaded_post_model, LinearTransform)
+    # post aggregation model should be the same as model_2
+    assert torch.equal(model_2.linear.weight, loaded_post_model.linear.weight)
+
+    checkpoint_module.maybe_checkpoint(model_2, 0.88, {"test_1": 1.0}, CheckpointMode.PRE_AGGREGATION)
+    checkpoint_module.maybe_checkpoint(model_1, 0.68, {"test_1": 1.0}, CheckpointMode.POST_AGGREGATION)
+    loaded_pre_model_best = pre_aggregation_checkpointer[0].load_best_checkpoint()
+    loaded_pre_model_latest = pre_aggregation_checkpointer[1].load_best_checkpoint()
+    loaded_post_model = post_aggregation_checkpointer.load_best_checkpoint()
+
+    assert isinstance(loaded_pre_model_best, LinearTransform)
+    # pre aggregation model should be the same as model_1 since the metric isn't better than the previous one
+    assert torch.equal(model_1.linear.weight, loaded_pre_model_best.linear.weight)
+
+    assert isinstance(loaded_pre_model_latest, LinearTransform)
+    # pre aggregation model should be the same as model_2
+    assert torch.equal(model_2.linear.weight, loaded_pre_model_latest.linear.weight)
+
+    assert isinstance(loaded_post_model, LinearTransform)
+    # post aggregation model should be the same as model_1
+    assert torch.equal(model_1.linear.weight, loaded_post_model.linear.weight)
+
+
+def test_path_duplication_check(tmp_path: Path) -> None:
+    checkpoint_dir = tmp_path.joinpath("resources")
+    checkpoint_dir.mkdir()
+    pre_aggregation_checkpointer: List[TorchCheckpointer] = [
+        BestLossTorchCheckpointer(str(checkpoint_dir), "pre_agg_best.pkl"),
+        LatestTorchCheckpointer(str(checkpoint_dir), "pre_agg_best.pkl"),
+    ]
+    post_aggregation_checkpointer = BestLossTorchCheckpointer(str(checkpoint_dir), "post_agg.pkl")
```

Comment:
Can we add another small test here to check the error if we have same path for pre_aggregation_checkpointer and post_aggregation_checkpointer?

Reply:
Yup. Will do now.